src/lib/deployWorkers.ts

```diff
@@ -622,13 +633,89 @@ const validateWasmExist = async (
   }
 };
 
+const DEAL_TYPE: Deals[string] = {
+  dealId: "",
+  chainNetwork: "testnet",
+  chainNetworkId: 0,
+  dealIdOriginal: "",
+  definition: "",
+  timestamp: "",
+};
+
+const HOSTS_TYPE: Hosts[string] = {
+  definition: "",
+  installation_spells: [
+    {
+      host_id: "",
+      spell_id: "",
+      worker_id: "",
+    },
+  ],
+  relayId: "",
+  timestamp: "",
+};
+
 export const ensureAquaFileWithWorkerInfo = async (
-  workersConfig: WorkersConfigReadonly
+  workersConfig: WorkersConfigReadonly,
+  fluenceConfig: FluenceConfigReadonly
 ) => {
+  const allWorkerNames = Object.keys(fluenceConfig.workers ?? {});
+
+  const workerNameDealsNills = allWorkerNames.reduce<
+    Record<string, ReturnType<typeof makeOptional<Deals[string]>>>
+  >((acc, workerName) => {
+    acc[workerName] = makeOptional(null, DEAL_TYPE);
+    return acc;
+  }, {});
+
+  const workerDealOptionals = Object.entries(workersConfig.deals ?? {}).reduce<
+    Record<string, ReturnType<typeof makeOptional<Deals[string]>>>
+  >(
+    (acc, [workerName, deal]) => {
+      acc[workerName] = makeOptional<Deals[string]>(deal, DEAL_TYPE);
+      return acc;
+    },
+    { ...workerNameDealsNills }
+  );
+
+  const workerNameHostsNills = allWorkerNames.reduce<
+    Record<string, ReturnType<typeof makeOptional<Hosts[string]>>>
+  >((acc, workerName) => {
+    acc[workerName] = makeOptional(null, HOSTS_TYPE);
+    return acc;
+  }, {});
+
+  const workerHostsOptionals = Object.entries(workersConfig.hosts ?? {}).reduce<
+    Record<string, ReturnType<typeof makeOptional<Hosts[string]>>>
+  >(
+    (acc, [workerName, hosts]) => {
+      acc[workerName] = makeOptional<Hosts[string]>(hosts, HOSTS_TYPE);
+      return acc;
+    },
+    { ...workerNameHostsNills }
+  );
+
   await writeFile(
     await ensureFluenceAquaWorkersPath(),
     jsToAqua(
-      { deals: workersConfig.deals, hosts: workersConfig.hosts },
+      {
+        deals: makeOptional(
+          Object.keys(workerDealOptionals).length === 0
+            ? null
+            : workerDealOptionals,
+          {
+            [DEFAULT_WORKER_NAME]: makeOptional(null, DEAL_TYPE),
```

Comment:
Will it override `DEFAULT_WORKER_NAME` in `workerDealOptionals`?

Reply:
It will not. In case there are no workers at all in the config and no workers were deployed before - it will put nil in aqua and create a type definition for DEFAULT_WORKER_NAME. Now that I am thinking about it again I am not completely sure that's the correct behaviour. I think I better not generate anyhing at all in this case instead of generating something, in a way, arbitrary. Thank you very much for your curiosity